src/Vulkan/RHI/VkSwapChain.cpp

```diff
@@ -34,15 +33,30 @@ static bool IsSwapChainSupported(::vk::PhysicalDevice device, ::vk::SurfaceKHR s
     return GetSwapChainSupportDetails(device, surface).IsValid();
 }
 
-static ::vk::PresentModeKHR GetBestPresentMode(const VkSwapChainSupportDetails& details, bool useVSync)
+static ::vk::SurfaceFormatKHR GetBestSurfaceFormat(const VkSwapChainSupportDetails& details,
+                                                   ::vk::Format requestedFormat)
 {
-    // VSync means we should use eFifo which is available on most platforms.
-    // Source: https://stackoverflow.com/questions/36896021/enabling-vsync-in-vulkan
-    if (useVSync)
+    for (const ::vk::SurfaceFormatKHR& availableFormat : details.formats)
     {
-        return ::vk::PresentModeKHR::eFifo;
+        if (availableFormat.format == requestedFormat &&
+            availableFormat.colorSpace == ::vk::ColorSpaceKHR::eSrgbNonlinear)
+        {
+            return availableFormat;
+        }
     }
 
+    VEX_LOG(Fatal, "Format \"{}\" not supported", ::vk::to_string(requestedFormat));
+    return {};
+}
+
+// Looks for mailbox which allows to always take the most recent image. Falls back to FIFO. If VSync is off, we instead
+// take immediate mode.
+static ::vk::PresentModeKHR GetBestPresentMode(const VkSwapChainSupportDetails& details, bool useVSync)
+{
+    if (!useVSync)
+        return ::vk::PresentModeKHR::eImmediate;
```

Comment:
Are you sure about this change? https://stackoverflow.com/questions/36896021/enabling-vsync-in-vulkan

Reply:
I think we should still check for swapchain compatibility of the selected present mode.

Reply:
yep thats ok with me